openstack/db/v1/instances/requests.go

```diff
@@ -0,0 +1,209 @@
+package instances
+
+import (
+	"fmt"
+
+	"github.com/rackspace/gophercloud"
+	db "github.com/rackspace/gophercloud/openstack/db/v1/databases"
+	"github.com/rackspace/gophercloud/openstack/db/v1/users"
+	"github.com/rackspace/gophercloud/pagination"
+)
+
+// CreateOptsBuilder is the top-level interface for create options.
+type CreateOptsBuilder interface {
+	ToInstanceCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts is the struct responsible for configuring a new database instance.
+type CreateOpts struct {
+	// Either the integer UUID (in string form) of the flavor, or its URI
+	// reference as specified in the response from the List() call. Required.
+	FlavorRef string
+
+	// Specifies the volume size in gigabytes (GB). The value must be between 1
+	// and 300. Required.
+	Size int
+
+	// Name of the instance to create. The length of the name is limited to
+	// 255 characters and any characters are permitted. Optional.
+	Name string
+
+	// A slice of database information options.
+	Databases db.BatchCreateOpts
```

Comment:
@jrperritt `db.BatchCreateOpts` is the type that satisfies the [`db.CreateOptsBuilder` interface](https://github.com/jamiehannaford/gophercloud/blob/cloud-dbs/openstack/db/v1/databases/requests.go#L63) since it serialises each one and nests it in the JSON object. Do you want to get rid of the batch type? If so, what will be responsible for transforming the slice into a JSON object?